File: Src/Icm.Core.Tests/Reflection/MyExample.vb
Imports Icm.Reflection


Public Class MyExample
    Public myfield As String = "mytest"
    Public mynumber As Integer = 7

    Property MyProp As String = "mytestp"
    Property MyPropNumber As Integer = 11

    Property ArrayProp As String() = {"a", "b"}

    Property IndexedProp(a As Integer) As String
        Get
            Return "<<" & a & ">>"
        End Get
        Set(value As String)

        End Set
    End Property

    Property IndexedProp2(a As Integer, b As String) As String
        Get
            Return "<<" & a & ";" & b & ">>"
        End Get
        Set(value As String)

        End Set
    End Property

    Property DictPropStr As New Dictionary(Of String, String) From {
        {"key1", "value1"},
        {"key2", "value2"}
    }

    Property DictPropInt As New Dictionary(Of Integer, String) From {
        {1234, "value1i"},
        {4567, "value2i"}
    }

    Property DictPropDate As New Dictionary(Of Date, String) From {
        {DateSerial(2012, 1, 2), "value1d"},
        {DateSerial(2013, 4, 7), "value2d"}
    }

End Class

File: Src/Icm.Core.Tests/Reflection/MyExample.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm.Reflection;


public class MyExample
{
	public string myfield = "mytest";

	public int mynumber = 7;
	public string MyProp { get; set; }
	public int MyPropNumber { get; set; }

	public string[] ArrayProp { get; set; }

	public string IndexedProp {
		get { return "<<" + a + ">>"; }

		set { }
	}

	public string IndexedProp2 {
		get { return "<<" + a + ";" + b + ">>"; }

		set { }
	}

	public Dictionary<string, string> DictPropStr { get; set; }

	public Dictionary<int, string> DictPropInt { get; set; }

	public Dictionary<System.DateTime, string> DictPropDate { get; set; }

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
